boards: arm: introduce support for ST Nucleo F722ZE

ryukoposting opened the pull request:
Nucleo F722ZE is a development board for the STM32F722ZE, one of ST's higher-end MCU offerings. Add support for the board, aligning with conventions set by similar boards already in the Zephyr distribution.

Some notes:
- `nucleo_f722ze`'s pinout is virtually identical to the existing `nucleo_f446ze` board.
- The `STM32F722XX` SoC is almost identical to the existing `STM32F723XX` SoC. The only difference is in the USB PHY peripherals.
- I am not affiliated with ST in any capacity.

I have tested several sample programs on my Nucleo F722ZE, including:

- samples/hello_world
- samples/basic/blinky
- samples/basic/button
- samples/drivers/watchdog
- samples/sensor/die_temp_polling
- samples/sensor/soc_voltage
- samples/subsys/shell/devmem_load (with `-DCONFIG_HWINFO_SHELL` added)

I have also run the above samples using the USB CDC-ACM and Segger RTT as consoles.

I have also tested these samples, though they require overlay files that aren't included in this PR (I'll add those later):

- samples/basic/fade_led
- samples/drivers/adc
- samples/drivers/dac

ryukoposting commented:
Apologies for the repeated force-pushes to get this thing going. I'm hoping this will be the last one - just a couple of tweaks to indentation in devicetree files.

erwango reviewed (changes requested):
LGTM aside one comment.
Also, please split the change in order to at least separate the tests changes from the board/soc additions. (Ideally also split board and soc additions but I won't block for this).

ryukoposting commented:
> LGTM aside one comment. Also, please split the change in order to at least separate the tests changes from the board/soc additions. (Ideally also split board and soc additions but I won't block for this).

Okay @erwango ! Just to make sure I understand properly, I _should_ have three PRs open:

1. The SoC definition (you're fine with me combining this with PR 2)
2. The board definition
3. Test overlay files for the board

Here's the thing... if I separate the test overlay files from the board definition, then tests won't pass on the second PR. Is that okay? Should I do something to force the tests to pass?

ryukoposting committed:
dts: arm: Add support for STM32F722 SoC

The STM32F722 is similar to the STM32F723, but lacks the latter's
more advanced USB PHY. Otherwise, they are virtually identical.

Signed-off-by: Evan Perry Grove <evan@4grove.com>

ryukoposting committed:
boards: arm: Add support for ST NUCLEO F722ZE

Introduce support for ST Nucleo F722ZE, an evaluation board based on
the STM32F722ZET6U microcontroller.

Signed-off-by: Evan Perry Grove <evan@4grove.com>

ryukoposting committed:
tests: drivers: Add nucleo_f722ze board support

Add overlays for the adc_api and counter_basic_api tests. Support
Nucleo F722ZE in the dac_api test.

Signed-off-by: Evan Perry Grove <evan@4grove.com>
